packages/block-editor/src/components/block-title/index.js

```diff
@@ -40,5 +53,17 @@ export default function BlockTitle( { clientId } ) {
 		return null;
 	}
 
-	return blockType.title;
+	const { __experimentalLabel, title } = blockType;
+
+	// Check if it supports the label first. getBlockLabel will return the title
+	// if no label exists. We want to know if the label is undefined so that we
+	// can format it uniquely.
+	const label =
+		__experimentalLabel &&
+		truncate( getBlockLabel( blockType, attributes ), { length: 15 } );
+
+	if ( label ) {
```

Comment:
I've noticed that some blocks define the `__experimentalLabel` but only return it in accessibility context.
When the label is falsy, [it defaults to the block title](https://github.com/WordPress/gutenberg/blob/4464bb36ac297295350127f6759303e39e9f22a8/packages/blocks/src/api/utils.js#L132-L153), which, combined to this PR, yield this result:

<img width="621" alt="Screenshot 2020-07-15 at 12 11 50" src="https://user-images.githubusercontent.com/2070010/87538604-629ae800-c694-11ea-9b17-601c460f1577.png">

We should check if the label is the same as the title before outputting both.
Something like:

```js
const { __experimentalLabel, title } = blockType;
const label = __experimentalLabel && getBlockLabel( blockType, attributes );

if ( label && label !== title ) {
  return `${ title } - ${ truncate( label, { length: 15 } ) }`;
}
return title;
```

Reply:
Good idea! That's a lot better than just checking if `__experimentalLabel` exists.